### ISSUE TITLE
508-defect-1 ❗Service periods can be opened and submitted blank

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#43258]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Details

The service period page has several critical issues
- Users with low vision may be unable to see the low contrast disabled button; they may become confused about why they can only input one service period at the given moment
- Users can submit multiple blank service periods (see video below) due to a lack of error messaging

## Acceptance Criteria

- [ ] Buttons are not disabled
- [ ] Descriptive error messaging is provided
- [ ] Users cannot submit blank service periods

## Steps to Recreate

1. [Go to step 3, service periods in the burials form](https://www.va.gov/burials-and-memorials/application/530/military-history/service-periods)
2. Confirm the add service button is disabled and fails to meet contrast requirements (new design system guidance explicitly states not to disable buttons, error messaging is recommended instead)
3. Type "a" into the first field
4. Confirm the add service button is enabled
5. Continue adding multiple service periods
6. Continue to the next page and confirm no error message is thrown

## Proposed Solution (if known)

I have a gut feeling that the need to make this page without validation stemmed from trying to accommodate users submitting via DD214. Asking a qualifying question upfront e.g. "Do you have a DD214?" should theoretically resolve the source of the issue. 

## WCAG or Vendor Guidance (optional)

[VA Design System: Do not disable buttons](https://design.va.gov/components/button/#do-not-disable-buttons)

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/175338255-1a67a8ce-6923-4e7b-ba36-889419c7da02.mov

**Comment by skylerschain** [comment]
The [blocker for this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/43256#issue-1282593416) is now "In Progress." Once that gets completed, it looks like we can resolve this by introducing some error messages for required fields.

**Comment by fiorella-io** [comment]
This defect will be fixed in the redesigned Burial form. It'll follow Skyler's proposed design but with the upload happening on step 5.

Design:
- [New DD214 question](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/v/1RQjx5/a/4oEjymo)
- [New DD214 document upload section on step 5](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/OeW2Mwq)

We might change the upload right after the question depending on the amount of effort it would take
